x-pack/plugins/siem/server/lib/timeline/saved_object.ts

```diff
@@ -94,12 +95,31 @@ export const getTimelineByTemplateTimelineId = async (
   return getAllSavedTimeline(request, options);
 };
 
+/** The filter here is able to handle the legacy data,
+ * which has no timelineType exists in the savedObject */
+const getTimelineTypeFilter = (timelineTypes: string | null) => {
+  if (timelineTypes === TimelineType.template) {
+    /** Show only whose timelineType is "template" */
+    return `siem-ui-timeline.attributes.timelineType: ${TimelineType.template}`;
+  }
+
+  if (timelineTypes === TimelineType.default) {
+    /** Show me every timeline whose timelineType is not "template".
+     * which includes timelineType === 'default' and
+     * those timelineType doesn't exists */
+    return `not siem-ui-timeline.attributes.timelineType: ${TimelineType.template}`;
+  }
+
+  return undefined;
```

Comment:
I do not think that think we should return undefined but always returning the default template since template timeline are edgecases
```
return `not siem-ui-timeline.attributes.timelineType: ${TimelineType.template}`;
```